src/mesh/surface_interpolant.jl

```diff
@@ -1,53 +1,45 @@
 # """
-#     GammaCurve{RealT<:Real, NNODES}
+#     CurvedSurface{RealT<:Real, NNODES}
 #
 # Contains the data needed to represent a curve with data points (x,y) as a Lagrange polynomial
 # interpolant written in barycentric form at a given set of nodes.
 #
 # !!! warning "Experimental code"
 #     This mesh type is experimental and can change any time.
 # """
-struct GammaCurve{RealT<:Real, NNODES}
-  nodes               ::SVector{NNODES, RealT}
-  barycentric_weights ::SVector{NNODES, RealT}
-  x                   ::SVector{NNODES, RealT}
-  y                   ::SVector{NNODES, RealT}
-end
-
-
-# construct a single instance of the gamma curve struct
-function GammaCurve(RealT, curve_polydeg, curve_x_vals, curve_y_vals)
-  nnodes_ = curve_polydeg + 1
-
-  nodes, _ = chebyshev_gauss_lobatto_nodes_weights(nnodes_)
-  wbary    = barycentric_weights(nodes)
-
-  return GammaCurve{RealT, nnodes_}(nodes, wbary, curve_x_vals, curve_y_vals)
+struct CurvedSurface{RealT<:Real, NNODES}
+  nodes               ::Vector{RealT}
+  barycentric_weights ::Vector{RealT}
+  coordinates         ::Array{RealT, 2}
 end
```

Comment:
It never uses it (it used to in a previous iteration). We can probably ditch it because the mesh store the `mesh_polydeg` such that this information is known to the user

Reply:
Yes, that would be good. If we don't use it, we shouldn't have it here.